mypy/sametypes.py

```diff
@@ -40,6 +40,20 @@ def simplify_union(t: Type) -> ProperType:
     return t
 
 
+def separate_union_literals(t: UnionType) -> Tuple[Sequence[LiteralType], Sequence[Type]]:
+    literal_items = []
+    union_items = []
+
+    for item in t.items:
+        proper = get_proper_type(item)
+        if isinstance(proper, LiteralType):
+            literal_items.append(proper)
+        else:
+            union_items.append(item)
+
+    return literal_items, union_items
+
```

Comment:
Maybe we can put this functions in the `typeop.py`? `sametypes` is not clear to me and there is a dicussion about it.  https://github.com/python/mypy/issues/7924